Documentation/Basic-Usage.md

```diff
@@ -128,7 +174,13 @@ extension User {
 }
 ```
 
-You could see how this would get much worse with a more complex model.
+Not only is that code much more verbose than the equivalent code using Argo,
+it's also not as safe. It does check to make sure that `id` and `name` are
+non-nil, but it's not checking to see that their types are correct. If that code
+encountered JSON where the `"id"` key returned, say a `String` instead of an
+`Int`, this would lead to a crash.
```

Comment:
I don't agree that mentioning type-safety is a straw-man argument (that's a very real thing that would likely even lead to a crash in objc code which might otherwise at least cope with a nil value), but the other points you raise are also important. I'll try to find some way to mention all of those.